src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -7112,6 +7112,130 @@ static void Main()
+            comp2.VerifyEmitDiagnostics();
+
+            comp2 = CreateCompilation(src2, references: [comp1.ToMetadataReference()], targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp2.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void Using_LanguageVersion_02()
+        {
+            var src1 = @"
+public ref struct S2 : System.IDisposable
+{
+    void System.IDisposable.Dispose()
+    {
+    }
+}
+";
+
+            var comp1 = CreateCompilation(src1, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+
+            var src2 = @"
+class C
+{
+    static void Main()
+    {
+        using (new S2())
+        {
+        }
+
+        using (var s = new S2())
+        {
+        }
+    }
+}
+";
+            var comp2 = CreateCompilation(src2, references: [comp1.ToMetadataReference()], targetFramework: s_targetFrameworkSupportingByRefLikeGenerics, parseOptions: TestOptions.Regular12);
+            comp2.VerifyEmitDiagnostics(
+                // (6,16): error CS8652: The feature 'ref struct interfaces' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         using (new S2())
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "new S2()").WithArguments("ref struct interfaces").WithLocation(6, 16),
+                // (10,16): error CS8652: The feature 'ref struct interfaces' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         using (var s = new S2())
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "var s = new S2()").WithArguments("ref struct interfaces").WithLocation(10, 16)
+                );
+
+            comp2 = CreateCompilation(src1 + src2, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics, parseOptions: TestOptions.Regular12);
+            comp2.VerifyEmitDiagnostics(
+                // (2,24): error CS8652: The feature 'ref struct interfaces' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                // public ref struct S2 : System.IDisposable
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "System.IDisposable").WithArguments("ref struct interfaces").WithLocation(2, 24)
+                );
+
+            comp2 = CreateCompilation(src2, references: [comp1.ToMetadataReference()], targetFramework: s_targetFrameworkSupportingByRefLikeGenerics, parseOptions: TestOptions.RegularNext);
+            comp2.VerifyEmitDiagnostics();
+
+            comp2 = CreateCompilation(src2, references: [comp1.ToMetadataReference()], targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp2.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void Using_LanguageVersion_04()
```

Comment:
> Nit: straight from 2 to 4?

There is no requirement to avoid gaps in numbers. There is a reason, however, the number matches the other test that covers similar scenario.